.github/workflows/ci.yml

```diff
@@ -66,6 +66,32 @@ jobs:
             ninja-build \
             pkg-config
 
+
+      - name: Cache uncrustify
+        id: cache-uncrustify
+        uses: actions/cache@v2
+        with:
+          path: ${{ env.CACHE_UNCRUSTIFY }}
+          key: ${{ env.UNCRUSTIFY_VERSION }}
+
+      - name: Clone uncrustify
+        if: steps.cache-uncrustify.outputs.cache-hit != 'true'
+        uses: actions/checkout@v2
+        with:
+          repository: uncrustify/uncrustify
+          ref: ${{ env.UNCRUSTIFY_VERSION }}
+          path: uncrustify
+
+      - name: Install uncrustify
+        if: steps.cache-uncrustify.outputs.cache-hit != 'true'
+        run: |
+          source_dir=uncrustify
+          build_dir=uncrustify/build
+          cmake -S $source_dir -B $build_dir -G Ninja -DCMAKE_BUILD_TYPE=Release
+          cmake --build $build_dir
+          mkdir -p $HOME/.cache
+          cp $build_dir/uncrustify ${{ env.CACHE_UNCRUSTIFY }}
```

Comment:
is this actually enough? is it building a static binary? otherwise one can use `CMAKE_INSTALL_PREFIX`

Reply:
> is it building a static binary?

Uh, I don't know. Does it matter?

> otherwise one can use CMAKE_INSTALL_PREFIX

Hmm, can you show how?

Reply:
Turned it out that it doesn't matter, unless you need the manpage 😄 

here's how to do it tho, in case they change the requirement in the future

```console
cmake --install $build_dir --strip --prefix ~/.cache/uncrustify # or ${{ env.CACHE_UNCRUSTIFY }}
~/.cache/uncrustify/bin/uncrustify --version
```

Reply:
What requirement? I'm so confused.

Do you mean I should make this change now or in the future when something changes?

Reply:
it's up to you, this is just a standard procedure and is a benefit of using cmake, it ensures:
1. the binary is available under `<prefix>/bin/uncrustify`, so no need to manually locate it in `$build_dir`
2. if uncrustify starts linking against a library and/or starts expecting headers, then copying the binary alone wouldn't work anymore

Reply:
Hmm, I see. OK, I'll test it out, thanks for the tip.